pkg/batch/executor_test.go

```diff
@@ -11,6 +11,50 @@ import (
 	"github.com/treeverse/lakefs/pkg/logging"
 )
 
+const accessedOnce = 1
+const accessedTwice = 2
+
+type reader struct {
+	Ch         chan bool
```

Comment:
Also, AFAICT this could even be
```suggestion
	Ch         chan struct{}
```
since we only ever close it.